### PR TITLE
Warn when 'self' is not found in contact points

### LINKED ISSUE #189: Fail when 'self' is not included in the service discovery
When service discovery does not discover the 'current' node (but none at all or only 'peers'), this seems to point to a misconfiguration (#187, https://github.com/akka/akka-management/issues/183#issuecomment-385932241).

Should we (retry and) fail in such a situation?

### PR DESCRIPTION
**Pull request by raboof**
fixes #189

### CODE REVIEW

File: cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala [hunk 1]
```diff
@@ -139,17 +139,25 @@ class LowestAddressJoinDecider(system: ActorSystem, settings: ClusterBootstrapSe
       def hostMatches(host: String, lowest: ResolvedTarget): Boolean =
         (host == lowest.host) || (Some(host) == lowest.address.map(_.getHostAddress))
 
-      // we check if a contact point is "us", by comparing host and port that we've bound to
-      def lowestContactPointIsSelfManagement(lowest: ResolvedTarget): Boolean = lowest.port match {
+      def matchesSelf(target: ResolvedTarget): Boolean = target.port match {
         case None =>
-          selfContactPoints.exists { case (host, _) => hostMatches(host, lowest) }
+          selfContactPoints.exists { case (host, _) => hostMatches(host, target) }
```
Comment (chbatey) [inline cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala]: Not this PR but very odd that selfContactPoints refers directly to Kubernetes API configuration. Users may have that but not be using kubernetes api for boostrap

File: cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala [hunk 2]
```diff
@@ -139,17 +139,25 @@ class LowestAddressJoinDecider(system: ActorSystem, settings: ClusterBootstrapSe
       def hostMatches(host: String, lowest: ResolvedTarget): Boolean =
         (host == lowest.host) || (Some(host) == lowest.address.map(_.getHostAddress))
```
Comment (chbatey) [inline cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala]: could be a contains

File: cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala [hunk 3]
```diff
@@ -139,17 +139,25 @@ class LowestAddressJoinDecider(system: ActorSystem, settings: ClusterBootstrapSe
...
       }
 
       lowestAddressContactPoint(info) match {
-        case Some(lowest) => lowestContactPointIsSelfManagement(lowest)
-        case None => false
+        case Some(lowest) =>
+          if (matchesSelf(lowest)) true
+          else {
+            if (!info.contactPoints.exists(matchesSelf)) {
+              log.warning("Self contact point [{}] not found in targets [{}]", selfContactPoints.mkString(", "),
```
Comment (chbatey) [inline cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala]: We could be a bit more explicit that this is likely a misconfiguration, up to you

File: cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala (hunk 1 above)
Comment (raboof) [reply cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/LowestAddressJoinDecider.scala]: Yes, I agree but didn't find another way - (https://github.com/akka/akka-management/pull/238#discussion_r201627109)

**Pull request merged**